api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubCommitV3.java

```diff
@@ -0,0 +1,318 @@
+
+        List<Commit> commitList = getCommits(commitsObjectList, repoUrl, branch, authorLDAPDN);
+
+        commitRepository.save(commitList);
+
+        return result;
+    }
+
+    protected List<Commit> getCommits(List<Map> commitsObjectList, String repoUrl,
+                                      String branch, String authorLDAPDN) throws MalformedURLException, HygieiaException, ParseException {
+        List<Commit> commitsList = new ArrayList<>();
+
+        GitHubParsed gitHubParsed = new GitHubParsed(repoUrl);
+        CollectorItem collectorItem = getCollectorItem(gitHubParsed.getUrl(), branch);
+        GitHubWebHookSettings gitHubWebHookSettings = parseAsGitHubWebHook(apiSettings.getGitHubWebHook());
+
+        List<Pattern> commitExclusionPatterns = new ArrayList<>();
+        if (!StringUtils.isEmpty(gitHubWebHookSettings.getNotBuiltCommits())) {
+            if (!StringUtils.isEmpty(gitHubWebHookSettings.getNotBuiltCommitsDelimiter())) {
+                String[] notBuiltCommits = gitHubWebHookSettings.getNotBuiltCommits().split(gitHubWebHookSettings.getNotBuiltCommitsDelimiter());
+                Arrays.stream(notBuiltCommits).map(regExStr -> Pattern.compile(regExStr, Pattern.CASE_INSENSITIVE)).forEach(commitExclusionPatterns::add);
+            } else {
+                commitExclusionPatterns.add(Pattern.compile(gitHubWebHookSettings.getNotBuiltCommits(), Pattern.CASE_INSENSITIVE));
+            }
+        }
+
+        for (Map cObj : commitsObjectList) {
+            Object repoMap = cObj.get("repository");
+            boolean isPrivate = restClient.getBoolean(repoMap, "private");
+            String repoToken = getRepositoryToken(repoUrl);
+            String gitHubWebHookToken =  isPrivate ? repoToken : gitHubWebHookSettings.getToken();
+
+            if (StringUtils.isEmpty(gitHubWebHookToken)) {
+                throw new HygieiaException("Failed processing payload. Missing Github API token in Hygieia.", HygieiaException.INVALID_CONFIGURATION);
+            }
+
+            Commit commit = new Commit();
+
+            String commitId = restClient.getString(cObj, "id");
+            commit.setScmRevisionNumber(commitId);
+
+            Object authorObjectFromCommit = cObj.get("author");
+            String authorName = restClient.getString(authorObjectFromCommit, "name");
+            commit.setScmAuthor(authorName);
+
+            String message = restClient.getString(cObj, "message");
+            commit.setScmCommitLog(message);
+
+            commit.setTimestamp(System.currentTimeMillis());
+            commit.setScmUrl(repoUrl);
+            commit.setScmBranch(branch);
+            commit.setScmAuthorLDAPDN(authorLDAPDN);
+
+            DateTime commitTimestamp = new DateTime(restClient.getString(cObj, "timestamp"));
+            DateTime commitTimestampStepBack = commitTimestamp.minusMinutes(gitHubWebHookSettings.getCommitTimestampOffset());
+
+            commit.setScmCommitTimestamp(commitTimestamp.getMillis());
```

Comment:
Shouldn't this be fetched from payload? Consider a scenario where github sends the payload late due to issues.